plugins/BEdita/API/tests/IntegrationTest/UniqueNameTest.php

```diff
@@ -0,0 +1,79 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2017 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\API\Test\IntegrationTest;
+
+/**
+ * Test on `uname` field
+ */
+class UniqueNameTest extends ApiIntegrationTestCase
+{
+    /**
+     * Data provider for testDoubleInsert
+     *
+     * @return void
+     */
+    public function doubleInsertProvider()
+    {
+        return [
+            'sameTitle' => [
+                ['title' => 'test double test double'],
+            ],
+            'emptyTitle' => [
+                ['title' => ''],
+            ],
+            'empty' => [
+                []
+            ],
+        ];
+    }
+
+    /**
+     * Test inserting the same data two times for different object types.
+     *
+     * @return void
+     * @dataProvider doubleInsertProvider
+     * @coversNothing
+     */
+    public function testDoubleInsert($attributes)
+    {
+        $sendRequest = function ($type) use ($attributes) {
+            $data = [
+                'type' => $type,
+                'attributes' => $attributes,
+            ];
+            $endpoint = '/' . $type;
+            $requestBody = json_encode(compact('data'));
+            $this->configRequestHeaders('POST');
+
+            $this->post($endpoint, $requestBody);
```

Comment:
More general remark: should we remove data we insert in a unit test?
in some tests we expect e predefined number of objects (from fixtures data)
Here is not a problem, but do we need a rule for that? just wondering...